packages/core/content-manager/server/controllers/relations.js

```diff
@@ -186,39 +186,43 @@ module.exports = {
     }
 
     const queryParams = {
-      sort: mainField,
-      ...query,
-      fields: fieldsToSelect, // cannot select other fields as the user may not have the permissions
-      filters: {}, // cannot filter for RBAC reasons
+      select: fieldsToSelect,
     };
 
     if (!isEmpty(idsToOmit)) {
-      addFiltersClause(queryParams, { id: { $notIn: idsToOmit } });
-    }
-
-    // searching should be allowed only on mainField for permission reasons
-    if (_q) {
-      addFiltersClause(queryParams, { [mainField]: { $containsi: _q } });
+      queryParams.filters = { id: { $notIn: idsToOmit } };
     }
 
-    const subQuery = strapi.db.queryBuilder(sourceModel.uid);
-
-    const alias = subQuery.getAlias();
-
-    const knexSubQuery = subQuery
-      .where({ id, [`${alias}.id`]: { $notNull: true } })
-      .join({ alias, targetField })
-      .select(`${alias}.id`)
-      .getKnexQuery();
-
-    addFiltersClause(queryParams, { id: { $in: knexSubQuery } });
-
     if (MANY_RELATIONS.includes(attribute.relation)) {
-      ctx.body = await strapi.entityService.findPage(targetedModel.uid, queryParams);
+      const page = Number(query.page || 1);
+      const pageSize = Number(query.pageSize || 10);
+
+      queryParams.offset = Math.max(page - 1, 0) * pageSize;
+      queryParams.limit = pageSize;
+
+      const [results, count] = await Promise.all([
+        strapi.db
+          .query(sourceModelUid)
+          .load({ id }, targetField, { ...queryParams, ordering: 'desc' }),
```

Comment:
does the ordering have to be in desc ? 
Testing it I see frontend expects it to be in 'asc'

Reply:
@ronronscelestes In which order are you displaying relations?

I first add 7 relations  in order
![image](https://user-images.githubusercontent.com/20578351/193552637-2cda5247-1b4f-46d8-acc5-5263f706a99e.png)

Whenever I add a new relationship it is added to the end. But when I save, I see only the first 5 relationships in the reverse order as Pierre mentioned.
![image](https://user-images.githubusercontent.com/20578351/193553294-18d80393-fa9f-498c-9f9f-f32e8862e06e.png)


If we were to change the ordering in the backend. You would have to load to the last page to see the last relation you added. Is this the behaviour we want?

Reply:
that's a good question @Marc-Roig I'm starting to get confused
I thought we would show the latest relations, the latest one being on top, then we would add the new ones at the bottom - I didn't manage to find the right US to confirm it though

Reply:
handled on the front before saving
<img width="477" alt="image" src="https://user-images.githubusercontent.com/71838159/193554295-d0cebac6-5fc2-4df1-aa0d-452feab0c2b2.png">

what we receive after saving
<img width="487" alt="image" src="https://user-images.githubusercontent.com/71838159/193554387-c27b802a-0aac-425d-89c2-d72a03a8f66e.png">

Reply:
> I thought we would show the latest relations, the latest one being on top, then we would add the new ones at the bottom - I didn't manage to find the right US to confirm it though

Isn't what we have now with this PR? The latest relation is displayed on top but when we add a new one they are placed at the end.

Reply:
> handled on the front before saving <img alt="image" width="477" src="https://user-images.githubusercontent.com/71838159/193554295-d0cebac6-5fc2-4df1-aa0d-452feab0c2b2.png">
> 
> what we receive after saving <img alt="image" width="487" src="https://user-images.githubusercontent.com/71838159/193554387-c27b802a-0aac-425d-89c2-d72a03a8f66e.png">

The backend is returning it with the inverse order. (latest first)
But the frontend expects the inverse (oldest first) ( I think)

Reply:
My opinion:
Backend should order with oldest first ordering, but the frontend should first load the last page, not the first one.

Reply:
maybe I got it wrong though - I don't remember the ordering discussions between Pierre and Gustav
I'm not sure we would be able to fetch from the latest page, I think we should reverse it in front - let me take a deeper look

Reply:
yes, second option :)